### PR TITLE
Compute start and end offsets from timestamps

### PR DESCRIPTION
**Pull request by faillefer**
# Description

This PR is based on PR #126 by @nanic (Thanks for the work !)
Implements #121

I've added integration tests and updated the code so that '--exit' is not mandatory when using --from-timestamp
I've had to rework getOffsetBounds method to better manage possible options 
Also I removed the usage of '-1' value when returning offset when there's an error, because '-1' is a meaningful value for kafka (OffsetNewest) 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Documentation

- [x] the change is mentioned in the `## [Unreleased]` section of `CHANGELOG.md`
- [x] the configuration yaml was changed and the example config in `README.md` was updated
- [x] a usage example was added to `README.md`
- [x] tests for the changes have been implemented (see: [Testing your changes](https://github.com/deviceinsight/kafkactl/blob/main/.github/contributing.md#testing-your-changes))

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -17,3 +17,15 @@ func ContainsInt32(list []int32, element int32) bool {
 	}
 	return false
 }
+
+func StrArrayEqual(a, b []string) bool {
```
Comment (d-rk) [inline util/util.go:21]:
```suggestion
func StringArraysEqual(a, b []string) bool {
```

**Review by d-rk — CHANGES_REQUESTED** [review]
Generally the PR looks good :+1: 
I found mainly minor issues. The only thing that I like to be added is the parsing of different timestamp formats.

**Comment by faillefer** [comment]
Thanks for this very complete review and for the proposed improvements :smiley:. I think they are all implemented in the last commits. 
It would be helpful to update the readme.md with supported timestamp/date formats. But shall we use "examples" to describe them (as in Go time package) or a more standard pattern (like yyyy-MM-dd'T'HH:mm:ss.SSS)

**Comment by d-rk** [comment]
thank you for the PR @faillefer :+1:

**Pull request merged**